docs/data/migration/migration-pickers-v6/migration-pickers-v6.md

```diff
@@ -242,6 +242,100 @@ To keep the same behavior, you can replace it by `hasLeadingZerosInFormat`
+:::
+
+#### Move `inputRef` inside the props passed to the hook
+
+The field hooks now only receive the props instead of an object containing both the props and the `inputRef`.
+
+```diff
+- const { inputRef, ...otherProps } = props
+- const fieldResponse = useDateField({ props: otherProps, inputRef });
++ const fieldResponse = useDateField(props);
+```
+
+If you are using a multi input range field hook, the same applies to `startInputRef` and `endInputRef` params
+
+```diff
+- const { inputRef: startInputRef, ...otherStartTextFieldProps } = startTextFieldProps
+- const { inputRef: endInputRef, ...otherEndTextFieldProps } = endTextFieldProps
+
+  const fieldResponse = useMultiInputDateRangeField({
+    sharedProps,
+-   startTextFieldProps: otherStartTextFieldProps,
+-   endTextFieldProps: otherEndTextFieldProps,
+-   startInputRef
+-   endInputRef,
++   startTextFieldProps,
++   endTextFieldProps
+  });
+```
+
+#### Rename the ref returned by the hook to `inputRef`
+
+When used with the v6 TextField approach (where the input is an `<input />` HTML element), the field hooks return a ref that needs to be passed to the `<input />` element.
+This ref was previously named `ref` and has been renamed `inputRef` for extra clarity.
+
+```diff
+  const fieldResponse = useDateField(props);
+
+- return <input ref={fieldResponse.ref} />
++ return <input ref={fieldResponse.inputRef} />
+```
+
+If you are using a multi input range field hook, the same applies to the ref in the `startDate` and `endDate` objects
+
+```diff
+  const fieldResponse = useDateField(props);
+
+  return (
+    <div>
+-     <input ref={fieldResponse.startDate.ref} />
++     <input ref={fieldResponse.startDate.inputRef} />
+      <span>–</span>
+-     <input ref={fieldResponse.endDate.ref} />
++     <input ref={fieldResponse.endDate.inputRef} />
+    </div>
+  )
+```
+
+#### Restructure the API of `useClearableField`
+
+The `useClearableField` hook API has been simplified to now take a `props` parameter instead of a `fieldProps`, `InputProps`, `clearable`, `onClear`, `slots` and `slotProps` parameters.
+
+You should now be able to directly pass the returned value from your field hook (e.g: `useDateField`) to `useClearableField`
+
+```diff
+  const fieldResponse = useDateField(props);
+
+- const { InputProps, onClear, clearable, slots, slotProps, ...otherFieldProps } = fieldResponse
+- const { InputProps: ProcessedInputProps, fieldProps: processedFieldProps } = useClearableField({
+-   fieldProps: otherFieldProps,
+-   InputProps,
+-   clearable,
+-   onClear,
+-   slots,
+-   slotProps,
+- });
+-
+-  return <MyCustomTextField {...processedFieldProps} InputProps={ProcessedInputProps} />
+
++ const processedFieldProps = useClearableField(fieldResponse);
++
++ return <MyCustomTextField {...processedFieldProps} />
+```
+
+:::info
+If your custom field is based on one of the examples of the [Custom field](/x/react-date-pickers/custom-field/) page,
```

Comment:
Nitpick: Have you considered linking to `/x/react-date-pickers/custom-field/#commonly-used-custom-field`?
Anything before this section in the mentioned page does not seem to be relevant for this change. 🤔

Reply:
The structure changes in https://github.com/mui/mui-x/pull/10649 and this sections goes away (unless we decide otherwise ofc).
Not sure how to have a better link than the top of the page then.

But you are right, the beginning of the page is not related to what we are talking about here.

Reply:
Gotcha.
We can update the link whenever that page is refactored and maybe we'd have a better section by that time. 🤔 
WDYT about adding the following comment `// TODO v7: update to a more specific section link when the refactored field components are released`?

Reply:
`TODO` added on #10649 :+1: